pkg/arvo/app/hall.hoon

```diff
@@ -3017,6 +3020,10 @@
   |=  pax/path
   ^-  (quip move _+>)
   ?:  ?=([%sole *] pax)  ~&(%hall-no-sole !!)
+  ?:  ?=([%export *] pax)
+    :_  +>.$
+    [ost.bol %diff %export [%hall-v0 grams:(~(got by stories) ~.inbox)]]~
```

Comment:
I don't know much about hall's internal structures, but it seems plausible that we'd want to dump more than just the inbox. Maybe list the active subscriptions, and gently try to restore them on import?

/cc @Fang-

Reply:
`talk` centralizes all its data into the `~.inbox`. Landscape didn't always use the inbox for everything iirc. I have no idea what Modulo's Chat does.

What we export here depends entirely on what we say we care about/what we want to support. At the very least, we _should_ be exporting grams from all of the stories.  
If we care about carrying over configuration (which includes subscriptions), we need `shape.story`.  
Everything else in the `story` we can either rebuild from `grams`, or probably don't want to carry over anyway. ie sequence numbers for our subscriptions, presence/status for ships in the channel, etc.

Reply:
I did `~.inbox` because that's what the current load/save json export stuff does. I added a version tag to the exported state for hall so we can have backwards compatibility here.